compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2341,6 +2342,16 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
 
         setTypeOfVarRef(tupleDeStmt.varRef, data);
 
+        if (tupleDeStmt.expr.getKind() == LIST_CONSTRUCTOR_EXPR) {
```

Comment:
The error is logged for the following also. That's not necessary?

```ballerina
type Record record {|
    int a;
|};

public function main() {
    Record x;
    [x] = [{a: 1}];
}
```

Reply:
This is fixed now